GDTuber Documentation Update (#154)
quellus/GDTuber: wiki/Setup.md

# Quick Setup

>  1. Upon first time Opening GDTube you should see a window with a Transparent Background and an menu bar on the left side of the window. The menu bar contains the following items from top to bottom: 

Reviewer: ```suggestion
 1. Upon first time Opening GDTuber you should see a window with a Transparent Background and a menu bar on the left side of the window. The menu bar contains the following items from top to bottom: 
```

Author: Changed here: https://github.com/quellus/GDTuber/pull/154/commits/99af74fbb4045e498b515ae079d877f4bd548b0a

    - Visibility Button (Eyeball Icon): Toggle for making element visible and non-visible
    - Duplicate Button: Creates new duplicate the Screen Object Element.
    - Gizmo Icon: Allows for transform, resize and rotation of Screen Object Element. 
    - Delete (Trash can): Removes Screen Object Element.  

3. Select your microphone by clicking the `Input Device` and selecting your preferred device from the drop down menu choices 
    </br>   
    <img src="./doc_pictures/InputMenu.png" alt="image" width="400" height="auto"> 

4. Click the Gear Icon to open the image Screen Object Element settings pop up menu to upload your own custom avatar or images (for more information see the **[Image Setup](#image-setup)** section)
    </br>   
    <img src="./doc_pictures/ImageOptions.png" alt="image" width="300" height="auto"> 
5. Clicking the Gizmo Icon allows for `Screen Object Element` transformations of the image as follows: 
    - left click to move the object on screen
    - right click to rotate the object on screen
    - scroll wheel to scale the selected object on screen

## Image Setup

> Currently GDTube Supports four kinds of state for an image/avatar: **Neutral**, **Blinking**, **Talking**, **Talking + Blinking**

Reviewer: ```suggestion
Currently GDTuber Supports four kinds of state for an image/avatar: **Neutral**, **Blinking**, **Talking**, **Talking + Blinking**
```

Author: Fixed here: https://github.com/quellus/GDTuber/pull/154/commits/99af74fbb4045e498b515ae079d877f4bd548b0a

> Currently GDTube supports two ways of setting images to state, [Single Image](#single-image) mode where each state is set to single image file and [Multiple Images](#multiple-images) mode where states are contained on a single image sprite sheet. 

Reviewer: ```suggestion
Currently GDTuber supports two ways of setting images to state, [Single Image](#single-image) mode where each state is set to single image file and [Multiple Images](#multiple-images) mode where states are contained on a single image sprite sheet. 
```

Author: Fixed here: https://github.com/quellus/GDTuber/pull/154/commits/99af74fbb4045e498b515ae079d877f4bd548b0a

> Any image formats supported by Godot should work in GDTube (For a list of supported image formats see Godot Docs [here](https://docs.godotengine.org/en/stable/tutorials/assets_pipeline/importing_images.html))

Reviewer: ```suggestion
Any image formats supported by Godot should work in GDTuber (For a list of supported image formats see Godot Docs [here](https://docs.godotengine.org/en/stable/tutorials/assets_pipeline/importing_images.html))
```

</br>   
<img src="./doc_pictures/SpriteSheetMenu.png" alt="image" width="250" height="auto"> 

### Single Image

Allows you to set your avatar states via individual image files to a state. 

To use this mode uncheck the *Combined Avatar Image* Option. 

#### Single Image Setup 

1. After adding a `Screen Object Element` click the Settings Button (Gear Icon) to open the image configuration options. This should give you a menu with image state options and buttons stating `Change Image`. 

2. Clicking the change Image button for any of the states will open your systems file explorer for you to select the image you want to use. 

3. Navigate and select the image file and it should appear for that state.
</br>   
<img src="./doc_pictures/ImageOptions.png" alt="image" width="300" height="auto"> 

### Multiple Images

Allows you to set your avatar states via single image sprite sheet file. 

To use this mode check the *Combined Avatar Image* Option.

#### Multiple Image Setup 

##### Multiple Image Sprite Sheet Requirements 

> To show up correctly your sprite sheet must be a 1:1 ratio image taking up an even quadrant of the image for each state.  Top left will be the  **Neutral** state, top right will be that **Talking** state, bottom left will be the **Blinking State**, bottom right will be the **Talking + Blinking** state. 

Reviewer: This first sentence isn't right. It doesn't have to be a 1:1 ratio. The example sprite just happened to be that ratio. The sprite sheet can be any resolution. The 4 images just need to be all the same size.

Author: Updated the wording on this here: https://github.com/quellus/GDTuber/pull/154/commits/99af74fbb4045e498b515ae079d877f4bd548b0a

Let me know if that sound better.

</br>   
<img src="./doc_pictures/DefaultAvatar.png" alt="image" width="100" height="auto"> 

1. After adding a `Screen Object Element` click the Settings Button (Gear Icon) to open the image configuration options. Make sure *Combined Avatar Image* Option is checked. 
</br>   
<img src="./doc_pictures/SpriteSheetMenu.png" alt="image" width="250" height="auto"> 

2. Click the Change Image button to open your systems file explorer. 

3. Navigate and select the sprite sheet image file you'd like to use for the predefined avatar states. You should now see your sprite sheet being used for all states that are enabled. 

## OBS Integration Recommendation

1. Create New or Load a scene in OBS

> 2. Under sources click the add button and choose window capture

Reviewer: For most Windows users, the original OBS setup instructions were correct. It's better to use Game Capture because it will capture the transparency. I'm assuming you changed this to Window Capture because it's not available on Linux?

If that's the case, I think it would make sense to create 2 different instructions. One for Windows that has the old instructions and another for Linux.

Author: Fair enough, I've added the old instructions back here (https://github.com/quellus/GDTuber/pull/154/commits/99af74fbb4045e498b515ae079d877f4bd548b0a)  and added an "Other systems section"
